### ISSUE TITLE
SetupWizard fails because of trichrome resp. chromium crash

### ISSUE THREAD

**Issue by stefanhh0** [opengapps/opengapps#804]
**Kernel version:** 4.9.210
**Android version:** android-10.0.0_r21
**Opengapps package:** pico
**Used installation method:** In system.img integrated via aosp_build
**Build:** aosp_g8441-userdebug 10 QQ1A.200105.002 eng.stefan.20200130.195129

The installation used the gitlab contents from 30.01.2020 19:50 CET

When doing a full clean/flash including userdata.img the SetupWizard keeps crashing. This time the correct Version of the SetupWizard was active on the system.img and the crash seems to be coming from chromium library:

```
01-31 21:52:50.286  5242  5242 E chromium: [ERROR:icu_util.cc(177)] Invalid file descriptor to ICU data received.
01-31 21:52:50.287  5242  5242 F libc    : Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x70ca9d3c54 in tid 5242 (.android.gms.ui), pid 5242 (.android.gms.ui)
```

See as well logcat.log and the written tombstone. 

[logcat.log](https://github.com/opengapps/opengapps/files/4141439/logcat.log)
[tombstone_00.log](https://github.com/opengapps/opengapps/files/4141440/tombstone_00.log)

### Conclusion:

When using aosp_build building with following overrides:
```
GAPPS_VARIANT := pico
GAPPS_PRODUCT_PACKAGES += \
    Chrome

GAPPS_FORCE_WEBVIEW_OVERRIDES := true
GAPPS_FORCE_BROWSER_OVERRIDES := true
```

Trichrome support in aosp is provided by `TrichromeLibrary.apk` found in:
 https://gitlab.opengapps.org/opengapps/arm64/blob/master/app/com.google.android.trichromelibrary/29/nodpi/373018658.apk

Updating `TrichromeLibrary.apk` to version `394513638` fixes the problem.

`TrichromeLibrary.apk` should be updated alongside with Chrome in gitlab on a regular basis.

**Comment by stefanhh0** [comment]
There are many exceptions in the log before that complain about a missing permission:
```
java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_DEVICE_CONFIG
```

I will test if adding:
```
<privapp-permissions package="com.google.android.gms">
    <permission name="android.permission.WRITE_DEVICE_CONFIG" />
</privapp-permissions>
```
foxes the problem.

**Comment by stefanhh0** [comment]
It didn't help because the package manager is removing it again:
PackageManager: Un-granting permission android.permission.WRITE_DEVICE_CONFIG from package com.google.android.gms (protectionLevel=524802 flags=0xa0cabe45)

**Comment by stefanhh0** [comment]
Including GooglePermissionController seems to be as well no valid solution. Still Setup Wizard fails and  the `WRITE_DEVICE_CONFIG` Anyway I am not sure if the permission is the problem here.

Updated log with GooglePermissionController showing same behavior:
[logcat.log](https://github.com/opengapps/opengapps/files/4142035/logcat.log)

**Comment by stefanhh0** [comment]
The workaround to get past the SetupWizard is to have no network connection (removed the simcard + no WLAN)

Then trying to open webview shell results as well in an immediate app-crash. See the tombstone here: 
[tombstone_37.log](https://github.com/opengapps/opengapps/files/4142102/tombstone_37.log)

So I think the problem has nothing to do with the `WRITE_DEVICE_CONFIG` but has to do with Android SystemWebView and Trichrome. Chrome itself is not affected it just starts without problems and shows Websites as expected.

**Comment by stefanhh0** [comment]
Commenting out:
```
GAPPS_PRODUCT_PACKAGES += \
    Chrome

GAPPS_FORCE_WEBVIEW_OVERRIDES := true
GAPPS_FORCE_BROWSER_OVERRIDES := true
```

In my AOSP build solved the problem, however now it it is no longer possible to use trichrome but only the AOSP default: Android System WebView 74.0.3729.186.

@MarijnS95 do you have an idea why trichrome support when including opengapps that way is broken? 

I believe I can do following to get trichrome support back:
- Install `Android System Webview Beta` from the playstore 
- Rebuild with the aboves settings -> `Android System Webview` fails
- Switch to `Android System WebView Beta` in Developer settings

I'll try that and will report.

**Comment by MarijnS95** [comment]
@nezorflame As [asked earlier](https://gitlab.opengapps.org/opengapps/arm64/merge_requests/2#note_138) what was the reason for [rolling back](https://gitlab.opengapps.org/opengapps/arm64/commit/9a4033de4edb5e7923b06e25a3463d0fe76ab2a5) the Chrome APKs?
I locally run the latest versions similar to that MR, but after rolling them back to the versions in the available on `master` webview crashes with the exact same error. It is to be noted that the newer version I use match the version of `com.google.android.webview.beta`.

**Comment by stefanhh0** [comment]
Just confirming that the problem is fixed:

Including
```
GAPPS_PRODUCT_PACKAGES += \
    Chrome

GAPPS_FORCE_WEBVIEW_OVERRIDES := true
GAPPS_FORCE_BROWSER_OVERRIDES := true
```

Together with 
```
stefan@mars:~/android/apk$ aapt dump badging TrichromeLibrary.apk 
package: name='com.google.android.trichromelibrary' versionCode='394513638' versionName='79.0.3945.136' compileSdkVersion='29' compileSdkVersionCodename='10'
sdkVersion:'29'
targetSdkVersion:'29'
application-label:'Trichrome Library'
application-icon-160:'res/drawable/icon_webview.png'
application-icon-240:'res/drawable-hdpi-v4/icon_webview.png'
application-icon-320:'res/drawable-xhdpi-v4/icon_webview.png'
application-icon-480:'res/drawable-xxhdpi-v4/icon_webview.png'
application: label='Trichrome Library' icon='res/drawable/icon_webview.png'
feature-group: label=''
  uses-gl-es: '0x20000'
  uses-feature-not-required: name='android.hardware.touchscreen'
  uses-feature-not-required: name='android.software.leanback'
supports-screens: 'small' 'normal' 'large' 'xlarge'
supports-any-density: 'true'
locales: '--_--'
densities: '160' '240' '320' '480'
native-code: 'arm64-v8a'
alt-native-code: 'armeabi-v7a'
```

So the TrichromeLibrary.apk available in gitlab.opengapps.org should ideally always have the same version or at least should be close to the version of the included chrome.

**Comment by stefanhh0** [comment]
Hello @NicholasBuse, 

since Ilja seems currently not available, maybe you can do something about it? I think updating Trichrome as outlined in https://github.com/opengapps/opengapps/issues/804#issue-558381301 would just fix the problem.

Kind regards,
Stefan

**Comment by vhp8rc7p** [comment]
Excuse me,I am a newbie,and I want to know which should I flash,mine is OnePlus 7,LOS17,it was shipped with Android 9,and now it's 10,so I am confused

**Comment by AnggaSP** [comment]
@stefanhh0 
Issue has been fixed in the gitlab repo, you may want to try rebuilding it and see if it still crashing.

About your suggestion, optimally we should update everything but newer chrome use app bundles which unfortunately prevent us from updating it. So the only way is to revert WebView to older version that match TriChromeLibrary and Chrome.

**Comment by stefanhh0** [comment]
@AnggaSP 
Thanks for the fix, good job! I can confirm that after removing the updates from the PlayStore for TrichromeLibrary and Chrome on my phone and then rebuilding it with the downgraded WebView, so that all three packages WebView, TrichromeLibrary and Chrome are consistently on version: 74.0.3729.186 I can now start WebView Shell which shows as well that version.

I assume that this fixes as well the SetupWizard, I haven't tried though since it would require a user-data wipe which I currently don't want to do.

Afterwards, Google Chrome and Android System Webview are then updated via PlayStore to version 80.0.3987.87, after having installed those updates, WebView Shell shows the same updated version and works as expected, so that is fine as well.

There is still a security concern, since now opengapps is delivered with a vulnerable browser and system web view (See CVE fixes that have been introduced after 74.0.3729.186)

I don't know much of those app bundles. Are you planning to integrate current stable from chrome, webview and trichrome? If I can contribute with something just let me know.

Because of the security issue introduced by running outdated and vulnerable chromium stuff, I will open another issue.

Thanks again for fixing & kind regards
Stefan

**Comment by nezorflame** [comment]
We won't be able to deliver new versions of Chrome, Photos and Maps due to the app bundle mechanics, unless these are provided in the Pixel images.
